controllers/certificaterequest/certificaterequest_controller.go

```diff
@@ -390,14 +390,18 @@ func (r *CertificateRequestReconciler) createCertificateSecret(reqLogger logr.Lo
 
 // revokeCertificateAndDeleteSecret revokes certificate if it exists
 func (r *CertificateRequestReconciler) revokeCertificateAndDeleteSecret(reqLogger logr.Logger, cr *certmanv1alpha1.CertificateRequest) error {
-	//todo - actually delete secret when revoking
-
-	if SecretExists(r.Client, cr.Spec.CertificateSecret.Name, cr.Namespace) {
+	exists, err := SecretExists(r.Client, cr.Spec.CertificateSecret.Name, cr.Namespace)
+	if err != nil {
+		return fmt.Errorf("error checking if secret exists: %w", err)
+	}
+	if exists {
 		err := r.RevokeCertificate(reqLogger, cr)
 		if err != nil {
-			return err //todo - handle error from certificate missing
+			// TODO: handle error from certificate missing
+			return fmt.Errorf("error revoking certificate: %w", err)
 		}
 	}
+
```

Comment:
This is a point of personal preference, but I've found that when you have long if statements with a single return at the end it's better to negate and return for a shorter if statement.

For example, comparing the two code snippets:

```
	if exists {
		err := r.RevokeCertificate(reqLogger, cr)
		if err != nil {
			// TODO: handle error from certificate missing
			return fmt.Errorf("error revoking certificate: %w", err)
		}
	}
	return nil
```

versus:

```
	if !exists {
	    reqLogger.Info("Secret does not exist")
	    return nil
	}
	
	err := r.RevokeCertificate(reqLogger, cr)
	if err != nil {
		// TODO: handle error from certificate missing
		return fmt.Errorf("error revoking certificate: %w", err)
	}

	reqLogger.Info("Certificate successfully revoked")
	return nil
```

It's a few more lines now, but I personally find that returning early ends up making the code more clear and easier to maintain over time.